datumaro/plugins/data_formats/camvid.py

```diff
@@ -350,9 +350,14 @@ def apply(self):
                 else:
                     segm_list[item.id] = (image_path, "")
 
+                self._check_hash_key_existence(item)
```

Comment:
I'm sorry to keep going back and forth, but I'd like to delete this function for a while. This is because the number of annotations could be quite huge (COCO has 2.5 million labeled instances). Therefore, iterating over the annotations can be slow so much. Until we find the solution to minimize the impact on performance, in `datumaro==1.3.0`, I think that leaving the only way to save the hash keys as turning on `save_hashkey_meta` mannually, such as
```python
dataset.export(dst_dir, save_hashkey_meta=True)
```

Reply:
I applied this kinds of turning on for `save_hashkey_meta` in `test_hashkey.py` like `dataset.export(test_dir, fxt_data_format, save_media=True, save_hashkey_meta=True)`. I'm not sure to turn on `save_hashkey_meta` as default in `Exporter`. So I would appreciate it if you let me know if I have to do it in a different way.

Reply:
Implement `def __eq__()` here
https://github.com/openvinotoolkit/datumaro/blob/565b95d36df9bbd681247b7e24c64fe5acaf09e1/datumaro/components/annotation.py#L215-L218
, and also add a unit test for the HashKey comparison `e.g. assert hash_key_1 == hash_key_1`.

Reply:
Add `save_hashkey_meta=True` here
https://github.com/openvinotoolkit/datumaro/blob/develop/datumaro/cli/commands/explore.py#L122
, and add `dataset.export(dir, save_hashkey_meta=True)` guidance for hash key reuse here
https://github.com/openvinotoolkit/datumaro/blob/develop/docs/source/docs/level-up/intermediate_skills/10_data_exploration.rst?plain=1#L35